### PR TITLE
[Paralell] Prepare file processors for parallel run

### PR DESCRIPTION
**Pull request by TomasVotruba**
I'm working on parallel run in Rector - https://github.com/rectorphp/rector/issues/2971  :hugs: 

After this PR, https://github.com/rectorphp/rector-src/pull/1583 it is required to return jsonable value objects, so paralell process can merge them around :) 

This PR updates architecture here, so the diff is always returned :+1:

### CODE REVIEW

File: tests/FileProcessor/FlexForms/Fixture/flexforms.xml [hunk 1]
```diff
@@ -50,44 +50,43 @@
...
-    </sDEF>
-</sheets>
+                                </items>
+                            </config>
+                        </TCEforms>
+                    </settings.topNewsRestriction>
+                </el>
+            </ROOT>
+        </sDEF>
+    </sheets>
```
Comment (TomasVotruba) [inline tests/FileProcessor/FlexForms/Fixture/flexforms.xml:91]: Not sure why, but the Rector change fixed wrong indent here :)

File: src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php [hunk 2]
```diff
@@ -54,15 +65,27 @@ public function process(File $file, Configuration $configuration): void
...
         if ($newYaml === $yaml) {
-            return;
+            return $systemErrorsAndFileDiffs;
         }
 
-        $newFileContent = Yaml::dump($newYaml, 99);
+        $spaceCount = $this->yamlIndentResolver->resolveIndentSpaceCount($oldYamlContent);
+
+        $newFileContent = Yaml::dump($newYaml, 99, $spaceCount);
         $file->changeFileContent($newFileContent);
```
Comment (TomasVotruba) [inline src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php:74]: Before, the YAML was printed with default 2 space indent. That broke the tabs spacing and returned false positive.

The `YamlIndentResolver` service now makes sure the original indent is always kept. The regular expression might need some tweaks, but the main idea is finished :slightly_smiling_face:

Comment (TomasVotruba) [inline src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php:74]: Before, the YAML was printed with default 2 space indent. That broke the tabs spacing and returned false positive.

The `YamlIndentResolver` service now makes sure the original indent is always kept. The regular expression might need some tweaks, but the main idea is finished :slightly_smiling_face:

**Comment by TomasVotruba** [comment]
I'm mergin, as rector-src depends on packages to be with same parallel paradigm.
Do not relase new version yet, just in case :)  we need to test the main feature first.

File: src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php (hunk 2 above)
Comment (staabm) [reply src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php:74]: alternatively/additoinally the dumper could try to read the `.editorconfig` to get the expected whitespace config

Comment (sabbelasichon) [reply src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php:74]: Thanks a lot men. Looks promising.

**Pull request merged**